openhgnn/trainerflow/SeHGNN_trainer.py

```diff
@@ -0,0 +1,866 @@
+        parser.add_argument("--gama", type=float, default=10,
+                            help="parameter for the KL loss")
+        parser.add_argument("--start-stage", type=int, default=0)
+        parser.add_argument("--reload", type=str, default='')
+
+        return parser.parse_args(args)
+
+    def set_random_seed(self, seed=0):
+        random.seed(seed)
+        np.random.seed(seed)
+        torch.manual_seed(seed)
+        if torch.cuda.is_available():
+            torch.cuda.manual_seed(seed)
+
+    def get_n_params(self, model):
+        pp = 0
+        for p in list(model.parameters()):
+            nn = 1
+            for s in list(p.size()):
+                nn = nn * s
+            pp += nn
+        return pp
+
+    def hg_propagate(self, new_g, tgt_type, num_hops, max_hops, extra_metapath, echo=False):
+        for hop in range(1, max_hops):
+            reserve_heads = [ele[:hop] for ele in extra_metapath if len(ele) > hop]
+            for etype in new_g.etypes:
+                stype, _, dtype = new_g.to_canonical_etype(etype)
+
+                for k in list(new_g.nodes[stype].data.keys()):
+                    if len(k) == hop:
+                        current_dst_name = f'{dtype}{k}'
+                        if (hop == num_hops and dtype != tgt_type and k not in reserve_heads) \
+                                or (hop > num_hops and k not in reserve_heads):
+                            continue
+                        if echo: print(k, etype, current_dst_name)
+                        new_g[etype].update_all(
+                            fn.copy_u(k, 'm'),
+                            fn.mean('m', current_dst_name), etype=etype)
+
+            # remove no-use items
+            for ntype in new_g.ntypes:
+                if ntype == tgt_type: continue
+                removes = []
+                for k in new_g.nodes[ntype].data.keys():
+                    if len(k) <= hop:
+                        removes.append(k)
+                for k in removes:
+                    new_g.nodes[ntype].data.pop(k)
+                if echo and len(removes): print('remove', removes)
+            gc.collect()
+
+            if echo: print(f'-- hop={hop} ---')
+            for ntype in new_g.ntypes:
+                for k, v in new_g.nodes[ntype].data.items():
+                    if echo: print(f'{ntype} {k} {v.shape}')
+            if echo: print(f'------\n')
+
+        return new_g
+
+    def clear_hg(self, new_g, echo=False):
+        if echo: print('Remove keys left after propagation')
+        for ntype in new_g.ntypes:
+            keys = list(new_g.nodes[ntype].data.keys())
+            if len(keys):
+                if echo: print(ntype, keys)
+                for k in keys:
+                    new_g.nodes[ntype].data.pop(k)
+        return new_g
+
+    def check_acc(self, preds_dict, condition, init_labels, train_nid, val_nid, test_nid):
```

Comment:
We already have the functions in the openhgnn/task/node_classification.py. So, it should be better to use our built-in function instead of remaking one.